src/OSPSuite.Core/Domain/Services/FormulaTask.cs

```diff
@@ -211,13 +214,35 @@ public void ExpandDynamicFormulaIn(IContainer container)
             if (dynamicFormula.Criteria.IsSatisfiedBy(entityUsingFormula))
                throw new CircularReferenceInSumFormulaException(dynamicFormula.Name, entityUsingFormula.Name);
 
+            dynamicFormula.Criteria = updateDynamicFormulaCriteria(dynamicFormula, entityUsingFormula);
             entityUsingFormula.Formula = dynamicFormula.ExpandUsing(allFormulaUsable, _objectPathFactory, _objectBaseFactory);
          });
       }
 
+      private DescriptorCriteria updateDynamicFormulaCriteria(DynamicFormula formula, IUsingFormula usingFormula)
+      {
+         //we need to replace IN PARENT criteria with actual criteria matching the parent of the usingFormula
+         var criteria = formula.Criteria;
+         var allInParentTags = criteria.Where(x => x.IsAnImplementationOf<InParentCondition>()).ToList();
+         var parent = usingFormula.ParentContainer;
+         if (!allInParentTags.Any() || parent == null)
+            return criteria;
+
+         //we clone the criteria and remove all instances of InParentCondition. Then we add the criteria to the parent specifically
+         var modifiedCriteria = criteria.Clone();
+         allInParentTags.Each(x => modifiedCriteria.RemoveByTag<InParentCondition>(x.Tag));
+         if (modifiedCriteria.Operator != CriteriaOperator.And)
+            throw new OSPSuiteException(Error.InParentTagCanOnlyBeUsedWithAndOperator);
```

Comment:
actually we could throw the exception earlier, before cloning the criteria.

Reply:
done